packages/nextjs/src/app-router/server/utils.ts

```diff
@@ -9,7 +9,8 @@ export const buildRequestLike = () => {
       e &&
       'message' in e &&
       typeof e.message === 'string' &&
-      e.message.toLowerCase().includes('Dynamic server usage'.toLowerCase())
+      e.message.toLowerCase().includes('Dynamic server usage'.toLowerCase()) ||
```

Comment:
The logic here is wrong. For the first error message it essentially does this:

```js
e?.message === 'string' && etc
```

That check isn't happening for the new error message. You'll need to put the `||` into a parentheses to make sure that also the second error gets the pre checks

Reply:
jup forgot braces